src/metabase/feature_extraction/async.clj

```diff
@@ -45,24 +47,67 @@
   (when (running? job)
     (future-cancel (@running-jobs id))
     (swap! running-jobs dissoc id)
-    (db/update! ComputationJob id :status :canceled)))
+    (db/update! ComputationJob id :status :canceled)
+    (log/info (format "Async job %s canceled." id))))
+
+(defn- time-delta-seconds
+  [^java.util.Date a ^java.util.Date b]
+  (Math/round (/ (- (.getTime b) (.getTime a)) 1000.0)))
+
+(defn- fresh?
+  "Is the cached job still fresh?
+
+   Uses the same logic as `metabase.api.card`."
+  [{:keys [created_at updated_at]}]
+  (let [duration (time-delta-seconds created_at updated_at)
```

Comment:
I don't like the idea of `updated_at` being used to signify the end  of a job.

It means if we do things like update intermediate status in the future, we'll need to be aware that `updated_at` is being overloaded.